ibis/mapd/client.py

```diff
@@ -0,0 +1,326 @@
+
+    def __repr__(self):
+        return '<MapD {}>'.format(str(self))
+
+    @classmethod
+    def parse(cls, spec):
+        if spec.startswith('Nullable'):
+            return cls(spec[9:-1], nullable=True)
+        else:
+            return cls(spec)
+
+    def to_ibis(self):
+        return self.dtypes[self.typename](nullable=self.nullable)
+
+    @classmethod
+    def from_ibis(cls, dtype, nullable=None):
+        dtype_ = type(dtype)
+        if dtype_ in cls.ibis_dtypes:
+            typename = cls.ibis_dtypes[dtype_]
+        elif dtype in cls.ibis_dtypes:
+            typename = cls.ibis_dtypes[dtype]
+        else:
+            raise NotImplemented('{0} not Implemented'.format(dtype))
+
+        if nullable is None:
+            nullable = dtype.nullable
+        return cls(typename, nullable=nullable)
+
+
+class MapDCursor(object):
+    """Cursor to allow the MapD client to reuse machinery in ibis/client.py
+    """
+
+    def __init__(self, cursor):
+        self.cursor = cursor
+
+    def to_df(self):
+        if isinstance(self.cursor, Cursor):
+            col_names = [c.name for c in self.cursor.description]
+            result = pd.DataFrame(self.cursor.fetchall(), columns=col_names)
+        elif self.cursor is None:
+            result = pd.DataFrame([])
+        elif isinstance(self.cursor, pd.DataFrame):
+            result = self.cursor
+        elif GPUDataFrame is not None and isinstance(
+                self.cursor, GPUDataFrame
+        ):
+            result = self.cursor
+
+        return result
+
+    def __enter__(self):
+        # For compatibility when constructed from Query.execute()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
+
+
+class MapDQuery(Query):
+    """
+
+    """
+    def _fetch(self, cursor):
+        # check if cursor is a pymapd cursor.Cursor
+        return self.schema().apply_to(cursor.to_df())
+
+
+class MapDClient(SQLClient):
+    """
+
+    """
+    database_class = Database
+    sync_query = MapDQuery
+    dialect = MapDDialect
+
+    def __init__(
+        self, uri: str=None, user: str=None, password: str=None,
+        host: str=None, port: int=9091, dbname: str=None,
+        protocol: str='binary', execution_type: int=3
+    ):
+        """
+
+        :param uri:
+        :param user:
+        :param password:
+        :param host:
+        :param port:
+        :param dbname:
+        :param protocol:
+        :param execution_type:
+        """
+        self.uri = uri
+        self.user = user
+        self.password = password
+        self.host = host
+        self.port = port
+        self.dbname = dbname
+        self.protocol = protocol
+        self.execution_type = execution_type
+
+        self.con = pymapd.connect(
+            uri=uri, user=user, password=password, host=host,
+            port=port, dbname=dbname, protocol=protocol
+        )
+
+    @property
+    def _table_expr_klass(self):
+        return ir.TableExpr
```

Comment:
You *should* be able to delete this property. I believe it is provided in the parent class.